engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/nLayout/area/impl/TextCompositor.java

```diff
@@ -39,6 +39,8 @@ public class TextCompositor {
 	private FontInfo fontInfo;
 	private int runLevel;
 
+	private final String SHY_STRING = "\u00ad";
```

Comment:
A third time SHY is defined. Please move this to a central place (CONSTANTS). Please do not add STRING but make the name more clear SOFT_HYPHEN

Reply:
We could do that. However, I don't see the point adding an additional level of indirection and a reference to another class just to avoid a single-character constant. Practicality beats purity.

Reply:
Fine with me if there is no Constants class around.